pandas/core/base.py

```diff
@@ -940,17 +941,20 @@ def value_counts(self, normalize=False, sort=True, ascending=False,
         counts : Series
         """
         from pandas.core.algorithms import value_counts
-        from pandas.tseries.api import DatetimeIndex, PeriodIndex
         result = value_counts(self, sort=sort, ascending=ascending,
                               normalize=normalize, bins=bins, dropna=dropna)
 
-        if isinstance(self, PeriodIndex):
+        if isinstance(self, gt.ABCPeriodIndex):
```

Comment:
ideally I'd like to see all this type of logic either in the class itself or in ``pandas/core/algorithms.py/value_counts`` (or a function there)